deployments/container/Dockerfile

```diff
@@ -12,29 +12,22 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-ARG GOLANG_VERSION=1.22.6
-FROM nvcr.io/nvidia/cuda:12.9.1-base-ubi9 AS build
-
-RUN yum install -y \
-    wget make git gcc \
-     && \
-    rm -rf /var/cache/yum/*
-
-ARG GOLANG_VERSION=x.x.x
-RUN set -eux; \
-    \
-    arch="$(uname -m)"; \
-    case "${arch##*-}" in \
-        x86_64 | amd64) ARCH='amd64' ;; \
-        ppc64el | ppc64le) ARCH='ppc64le' ;; \
-        aarch64) ARCH='arm64' ;; \
-        *) echo "unsupported architecture" ; exit 1 ;; \
-    esac; \
-    wget -nv -O - https://storage.googleapis.com/golang/go${GOLANG_VERSION}.linux-${ARCH}.tar.gz \
-    | tar -C /usr/local -xz
-
-ENV GOPATH=/go
-ENV PATH=$GOPATH/bin:/usr/local/go/bin:$PATH
+FROM golang:1.24.6 AS base
+
+# The devel image is used for local development.
+FROM base AS devel
+
+WORKDIR /work
+COPY * .
+
+RUN make install-tools
+
+# We need to set the /work directory as a safe directory.
+# This allows git commands to run in the container.
+RUN git config --file=/.gitconfig --add safe.directory /work
```

Comment:
Why do we need to run git commands within the container?

Reply:
https://github.com/NVIDIA/k8s-device-plugin/blob/c87ad4047491d18999d666a9543bd0d134b86122/versions.mk#L30

Reply:
This is the `devel` image. Without this step go builds fail when running a `.shell` target.